src/System.Diagnostics.Process/tests/ProcessTestBase.cs

```diff
@@ -73,6 +73,18 @@ protected Process CreateProcess(Func<string, int> method, string arg)
             return p;
         }
 
+        protected Process CreateProcess(Func<string, int> method, string arg, ProcessStartInfo psi)
+        {
+            Process p = null;
+            using (RemoteInvokeHandle handle = RemoteInvoke(method, arg, new RemoteInvokeOptions { StartInfo = psi, Start = false }))
+            {
+                p = handle.Process;
+                handle.Process = null;
```

Comment:
Why is this helper needed?  And why are we nulling out the process in the handle?

Reply:
I believe if we don't null out the `handle.Process` here, then we cannot dispose the RemoteInvokeHandle here.
When we comment out the line `handle.Process = null` here we get InvalidOperationException during RemoteInvokeHandle.Dispose() operation.

Also note the two other existing CreateProcess() helper methods null out the process in handle as well. [Here]

These CreateProcess methods create and return the process through the handle but ProcessTestBase.Dispose() will later end up disposing the processes created this way.

[this diff](https://github.com/maryamariyan/corefx/commit/6ab836db8df5ec9af77063e45f42adccf86ec0b5#diff-18e45732507a229b452119e44e6d7f6c) shows the history of when they were added.
As you can see in this diff, previously CreateProcess() was not disposing of the RemoteInvokeHandle here and this its finalizer was Disposing of it. This is no longer accepted ever since we've added the `Assert.True(disposing);` statement in the RemoteInvokeHandle.Dispose(bool disposing) method.

Reply:
> When we comment out the line handle.Process = null here we get InvalidOperationException during RemoteInvokeHandle.Dispose() operation.

Why?  What is the exception due to?

Reply:
Previously we weren't disposing the handle and the finalizer was disposing it: 

```
Process p = RemoteInvoke(method, arg, new RemoteInvokeOptions { Start = false }).Process;
```

Reply:
```        
System.InvalidOperationException : No process is associated with this object.
        Stack Trace:
           c:\git\corefx\src\System.Diagnostics.Process\src\System\Diagnostics\Process.cs(903,0): at System.Diagnostics
  .Process.EnsureState(State state)
           c:\git\corefx\src\System.Diagnostics.Process\src\System\Diagnostics\Process.cs(913,0): at System.Diagnostics
  .Process.EnsureState(State state)
           c:\git\corefx\src\System.Diagnostics.Process\src\System\Diagnostics\Process.Windows.cs(793,0): at System.Dia
  gnostics.Process.GetProcessHandle(Int32 access, Boolean throwIfExited)
           c:\git\corefx\src\System.Diagnostics.Process\src\System\Diagnostics\Process.Windows.cs(149,0): at System.Dia
  gnostics.Process.WaitForExitCore(Int32 milliseconds)
           c:\git\corefx\src\System.Diagnostics.Process\src\System\Diagnostics\Process.cs(1326,0): at System.Diagnostic
  s.Process.WaitForExit(Int32 milliseconds)
           c:\git\corefx\src\CoreFx.Private.TestUtilities\src\System\Diagnostics\RemoteExecutorTestBase.cs(241,0): at S
  ystem.Diagnostics.RemoteExecutorTestBase.RemoteInvokeHandle.Dispose(Boolean disposing)
           c:\git\corefx\src\CoreFx.Private.TestUtilities\src\System\Diagnostics\RemoteExecutorTestBase.cs(228,0): at S
  ystem.Diagnostics.RemoteExecutorTestBase.RemoteInvokeHandle.Dispose()
           c:\git\corefx\src\System.Diagnostics.Process\tests\ProcessTestBase.cs(83,0): at System.Diagnostics.Tests.Pro
  cessTestBase.CreateProcess(Func`2 method, String arg, ProcessStartInfo psi)
           c:\git\corefx\src\System.Diagnostics.Process\tests\ProcessStartInfoTests.cs(360,0): at System.Diagnostics.Te
  sts.ProcessStartInfoTests.TestWorkingDirectoryPropertyInChildProcess
```

Reply:
The code doesn't need to be structured like that.  Why can't it be:
```C#
using (var h = RemoteInvoke(...))
{
    Process p = h.Process;
    ...
}
```
?

If there's really a reason, seems like that finalizer that got added is causing more problems than it's solving.